### PR TITLE
implemented refund service

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: HappyTravel.Edo.Data/Payments/Refund.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using HappyTravel.Money.Enums;
+
+namespace HappyTravel.Edo.Data.Payments
+{
+    public class Refund
```
Comment (mingazhev) [inline HappyTravel.Edo.Data/Payments/Refund.cs:6]: Credit card or ngenius refunds may be? They are used for ngenius only

Comment (DFSko) [reply HappyTravel.Edo.Data/Payments/Refund.cs]: renamed to NGeniusRefund

File: Api/Services/Payments/NGenius/NGeniusRefundService.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    join refund in _context.NGeniusRefunds on payment.Id equals refund.PaymentId
+                    where refund.PlannedDate >= date && payment.Status == PaymentStatuses.Captured
+                    select payment.Id)
+                .ToListAsync();
+        }
+
+
+        public async Task<Result<BatchOperationResult>> RefundPayments(List<int> paymentIds)
+        {
+            var payments = await (from payment in _context.Payments
```
Comment (mingazhev) [inline Api/Services/Payments/NGenius/NGeniusRefundService.cs:43]: Payments with refunds I guess

**Review by mingazhev — APPROVED** [review]
Merge only after local test on happy scenario.
+ small comment

File: Api/Services/Payments/NGenius/NGeniusRefundService.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Payments/NGenius/NGeniusRefundService.cs]: fixed

**Pull request merged**